src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -251,35 +257,37 @@ class DetailsCard extends React.Component {
       let fieldUpdated
       switch (fieldName) {
         case 'cluster':
-          updates = updates.set('cluster', this.props.clusters.get(value))
+          updates = updates.set('cluster', clusters.get(value))
           fieldUpdated = 'cluster'
 
           // Change the template to 'Blank' if the VM's template isn't in the new cluster
           {
-            const template = this.props.templates.get(updates.getIn(['template', 'id']))
+            const template = templates.get(updates.getIn(['template', 'id']))
             if (template && template.get('clusterId') && template.get('clusterId') !== value) {
-              changeQueue.push({ fieldName: 'template', value: this.props.blankTemplateId })
+              changeQueue.push({ fieldName: 'template', value: blankTemplateId })
             }
           }
           break
 
         case 'template':
-          updates = updates.set('template', this.props.templates.get(value))
+          updates = updates.set('template', templates.get(value))
           fieldUpdated = 'template'
 
           // Apply settings from the template to the VM (memory, CPUs, OS, cloudInit, bootMenuEnabled)
           {
             const template = this.props.templates.get(value)
             if (template) {
               const templateOsType = template.getIn(['os', 'type'], 'other')
-              const templateOs = this.props.operatingSystems.find(os => os.get('type') === templateOsType)
+              const oldOS = operatingSystems.find(os => os.get('name') === templateOsType)
+              const clustersOs = createOsList(updates.getIn(['cluster', 'id']), clusters, operatingSystems)
+              const templateOs = clustersOs.find(os => os.value === oldOS.get('description')) || clustersOs.find(os => os.value === 'Other OS')
```

Comment:
As mentioned  [here](https://github.com/oVirt/ovirt-web-ui/pull/1349#discussion_r545376619) I think we should base the logic on unique names/IDs.

Reply:
I've sent you question on slack on this subject, I'll post the conclusions here later